src/packing.cpp

```diff
@@ -23,66 +18,56 @@ namespace robothub {
-    llrc.do_kick = command.kicker;                                                  // [0, 1]          {true, false}
-    llrc.do_chip = command.chipper;                                                 // [0, 1]          {true, false}
-    llrc.kick_chip_forced = command.kicker_forced || command.chipper_forced;        // [0, 1]          {true, false}
+    llrc.do_kick = command.kicker();                                                  // [0, 1]          {true, false}
+    llrc.do_chip = command.chipper();                                                 // [0, 1]          {true, false}
+    llrc.kick_chip_forced = command.chip_kick_forced();        // [0, 1]          {true, false}
     llrc.kick_chip_power = (int) floor(kick_chip_power*255/8);                      // [0, 255]        [0, 100]%
-    llrc.velocity_dribbler = command.dribbler;                                      // [0, 31]        [0, 100]%
+    llrc.velocity_dribbler = command.dribbler();                                      // [0, 31]        [0, 100]%
 
-    llrc.geneva_drive_state = command.geneva_state;                                 // [(0)1, 5]       [-2, 2]
+    llrc.geneva_drive_state = command.geneva_state();                                 // [(0)1, 5]       [-2, 2]
     llrc.cam_position_x = 0;                                                        // [-4096, 4095]   [-10.24, 10.23]
     llrc.cam_position_y = 0;                                                        // [-4096, 4095]   [-10.24, 10.23]
     llrc.cam_rotation = 0;                                                          // [-1024, 1023]   [-pi, pi>
 
-    if (worldOpt) {
-        try {
-            std::shared_ptr<roboteam_msgs::WorldRobot> findBot = utils::getWorldBot(command.id, true, *worldOpt);
-            roboteam_msgs::WorldRobot robot;
-            if (findBot) {
-                robot = *findBot;
-                llrc.cam_position_x = 0;//(int) (robot.pos.x/10.24*4096); // set to 0 to avoid mystery stop bug
-                llrc.cam_position_y = 0;//(int) (robot.pos.y/10.24*4096); // set to 0 to avoid mystery stop bug
-                llrc.cam_rotation = (int) floor(robot.angle/M_PI*1024);
-                llrc.use_cam_info = true;
-            }
-
-        }
-        catch (std::out_of_range e) {
-            ROS_WARN_STREAM_THROTTLE(1, "[createLowLevelRobotCommand] Robot " << command.id
-                                                                              << " not present in World! Not adding camera data");
-        }
-        catch (std::exception e) {
-            ROS_WARN_STREAM_THROTTLE(1,
-                    "[createLowLevelRobotCommand] Something went wrong while adding camera data for robot "
-                            << command.id << " : " << e.what());
-        }
+
+    std::shared_ptr<roboteam_proto::WorldRobot> findBot = utils::getWorldBot(command.id(), true, worldOpt);
+  roboteam_proto::WorldRobot robot;
+    if (findBot) {
```

Comment:
This is problematic if the AI frame of reference is rotated but the robot is not since you're not rotating the world here. You would get a 180 degree wrong calibration in some situations.

Reply:
You don't see this in grSim because the calibration there is not done using this variable (we should probably change that)